pluginbase.py

```diff
@@ -68,3 +59,15 @@ def get_by_name(plugin_name: str) -> Plugin:
     if plugin is None:
         raise ValueError(f'Plugin "{plugin_name}" does not exist')
     return plugin
+
+
+if typing.TYPE_CHECKING:
```

Comment:
Is this only for making the linter happy? If so... I want to avoid this weirdness.

Reply:
it is. If you don't know what it means, you can ignore it.

But what you get by it, are useful linter checks that tell you when you do something wrong. And you should not ignore that.

Reply:
Let's talk in the other thread about removing the circular dependency.